pdns/pdnsutil.cc

```diff
@@ -4084,14 +4086,18 @@ try
       return 1;
     }
 
-    DNSBackend *src = nullptr;
-    DNSBackend *tgt = nullptr;
+    shared_ptr<DNSBackend> src{nullptr};
+    shared_ptr<DNSBackend> tgt{nullptr};
 
-    for(DNSBackend *b : BackendMakers().all()) {
-      if (b->getPrefix() == cmds.at(1))
-        src = b;
-      if (b->getPrefix() == cmds.at(2))
-        tgt = b;
+    for (auto& backend : BackendMakers().all()) {
+      auto* backendPtr = backend.release();
+
+      if (backendPtr->getPrefix() == cmds.at(1)) {
+        src = std::make_shared<DNSBackend>(backendPtr);
+      }
+      if (backendPtr->getPrefix() == cmds.at(2)) {
+        tgt = std::make_shared<DNSBackend>(backendPtr);
+      }
```

Comment:
@rgacogne Is this any better?

Reply:
Since we can abort when OLD == NEW then using `unique_ptr`s for `src` and `tgt` and moving them out should be fine.

Reply:
Nope, you would potentially have two smart pointers managing the same memory which is a terrible idea :)

Reply:
Ah correct. So it's `backendPtr` which should be the `shared_ptr` here.
Anyway, I switched them to `unique_ptr`.

Reply:
I'm not sure a shared pointer is the best solution. Peter confirmed that we can stop right away if the `OLD` and `NEW` backend names are the same, so if we add that early check we can just keep using unique pointers, unless I'm missing something.